bin/inx-chronicle/src/api/routes.rs

```diff
@@ -61,51 +61,59 @@ async fn login(
     }
 }
 
-async fn is_healthy(database: Extension<MongoDb>) -> bool {
-    let end = match database.find_last_milestone(u32::MAX.into()).await {
-        Ok(Some(last)) => last,
-        _ => return false,
-    };
-
-    // Panic: The milestone_timestamp is guaranteeed to be valid.
-    let latest_ms_time = OffsetDateTime::from_unix_timestamp(end.milestone_timestamp.0 as i64).unwrap();
-
-    if OffsetDateTime::now_utc() > latest_ms_time + STALE_MILESTONE_DURATION {
-        return false;
-    }
-
-    let start = match database.find_first_milestone(0.into()).await {
-        Ok(Some(first)) => first,
-        _ => return false,
-    };
-
-    // Check if there are no gaps in the sync status.
-    match database
-        .get_sync_data(start.milestone_index..=end.milestone_index)
-        .await
+async fn is_healthy(database: &MongoDb) -> bool {
+    #[cfg(feature = "stardust")]
     {
-        Ok(sync) => sync.gaps.is_empty(),
-        _ => false,
+        let end = match database.find_last_milestone(u32::MAX.into()).await {
+            Ok(Some(last)) => last,
+            _ => return false,
+        };
+
+        // Panic: The milestone_timestamp is guaranteeed to be valid.
+        let latest_ms_time = OffsetDateTime::from_unix_timestamp(end.milestone_timestamp.0 as i64).unwrap();
+
+        if OffsetDateTime::now_utc() > latest_ms_time + STALE_MILESTONE_DURATION {
+            return false;
+        }
+
+        let start = match database.find_first_milestone(0.into()).await {
+            Ok(Some(first)) => first,
+            _ => return false,
+        };
+
+        // Check if there are no gaps in the sync status.
+        match database
+            .get_sync_data(start.milestone_index..=end.milestone_index)
+            .await
+        {
+            Ok(sync) => sync.gaps.is_empty(),
+            Err(e) => {
+                log::error!("An error occured during health check: {e}");
+                return false;
+            }
+        };
     }
+
+    true
 }
 
 pub async fn info(database: Extension<MongoDb>) -> InfoResponse {
     InfoResponse {
         name: "Chronicle".into(),
         version: std::env!("CARGO_PKG_VERSION").to_string(),
-        is_healthy: is_healthy(database).await,
+        is_healthy: is_healthy(&database).await,
     }
 }
 
 pub async fn health(database: Extension<MongoDb>) -> StatusCode {
-    if is_healthy(database).await {
+    if is_healthy(&database).await {
         StatusCode::OK
     } else {
         StatusCode::SERVICE_UNAVAILABLE
     }
 }
 
-#[cfg(feature = "api-history")]
+#[cfg(all(feature = "stardust", feature = "api"))]
 pub async fn sync(database: Extension<MongoDb>) -> ApiResult<SyncDataDto> {
```

Comment:
This could be moved to the history folder, since that's where it's used.

Reply:
Done